src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -669,9 +671,10 @@ def __init__(self):
     def authenticate(self, username, password):
         try:
             user = ACCESS_CTRL_DB.get_user(username)
-            pass_hash = password_hash(password, user.password)
-            if pass_hash == user.password:
-                return user.permissions_dict()
+            if user.password:
```

Comment:
What happens if the password is either incorrect or empty?
What error/message do we return here?

Reply:
If no password is specified, user will see the same error that is displayed for incorrect password:

![screenshot from 2018-09-18 13-30-30](https://user-images.githubusercontent.com/14297426/45687645-3e792900-bb47-11e8-8a16-d33b647233fa.png)

Reply:
The user won't get this message in the current UI as the login page requires a password to be typed in before you can click on login.